src/Randomizer.SMZ3/AdvancedLogic.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Randomizer.Shared;
+using static Randomizer.SMZ3.SMLogic;
+
+namespace Randomizer.SMZ3
+{
+    public class AdvancedLogic
```

Comment:
One small remark: I'd pull these into an ILogic interface and use that on World instead, so we could swap out different "kinds" of logic more easily later on.